geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/cli/functions/LuceneListIndexFunction.java

```diff
@@ -56,12 +60,25 @@ public void execute(final FunctionContext context) {
     final Cache cache = context.getCache();
     final String serverName = cache.getDistributedSystem().getDistributedMember().getName();
     LuceneServiceImpl service = (LuceneServiceImpl) LuceneServiceProvider.get(cache);
-    for (LuceneIndex index : service.getAllIndexes()) {
-      indexDetailsSet.add(new LuceneIndexDetails((LuceneIndexImpl) index, serverName));
-    }
-
     for (LuceneIndexCreationProfile profile : service.getAllDefinedIndexes()) {
-      indexDetailsSet.add(new LuceneIndexDetails(profile, serverName));
+      indexDetailsSet
+          .add(new LuceneIndexDetails(profile, serverName, LuceneIndexStatus.NOT_INITIALIZED));
+    }
+    for (LuceneIndex index : service.getAllIndexes()) {
+      LuceneIndexStatus initialized = LuceneIndexStatus.INITIALIZED;
+      if (index instanceof LuceneIndexForPartitionedRegion) {
+        PartitionedRegion userRegion = (PartitionedRegion) cache.getRegion(index.getRegionPath());
```

Comment:
Maybe introduce a method on LuceneIndex to check the status of the index?  Then we could avoid the direct cast here and just call the index.getStatus() method.  Each index would know how to report it's status?